cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake

```diff
@@ -1005,6 +1020,42 @@ MACRO(TRIBITS_SETUP_INSTALLATION_PATHS)
 ENDMACRO()
 
 
+#
+# Generate resource spec file
+#
+
+
+MACRO(TRIBITS_GENERATE_RESOURCE_SPEC_FILE)
+  SET(GPUS_JSON)
+  MATH(EXPR LAST_GPU "${${PROJECT_NAME}_CUDA_NUM_GPUS} - 1")
+  SET(FIRST 1)
+  FOREACH(GPU RANGE 0 ${LAST_GPU})
+    IF(NOT FIRST)
+      STRING(APPEND GPUS_JSON ",\n")
+    ENDIF()
+    SET(FIRST 0)
+    STRING(APPEND GPUS_JSON "        {
+          \"id\": \"${GPU}\",
+          \"slots\": ${${PROJECT_NAME}_CUDA_SLOTS_PER_GPU}
+        }")
+  ENDFOREACH()
+  FILE(WRITE "${CMAKE_BINARY_DIR}/CMakeFiles/resources.json" "{
```

Comment:
Could we change the name of the generated file to `ctest_resources.json`?  That is a little more clear than `resources.json`.  Also, it might be good to move this to the base build directory so that it is not hidden away.  I tend to think of the `CMakeFiles` directory a place where CMake-proper writes files.

Reply:
Yes, that's fine.